### PR TITLE
[BATCH][V2] Marked V2 Metadata as Init instead of open session

### PR DESCRIPTION
**Pull request by zwangsheng**
### _Why are the changes needed?_

When we try to recover all running tasks before the restart on a specified Kyuubi service, we encounter that the kyuubi server is booted in  Start -> OOM  -> Start vicious cycle when there are too many tasks.

Here we set all need recover task as `INITIALIZED`, let every kyuubi service have chance to pick and run those job, to avoid the problem of excessive single point of load

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_
No

### CODE REVIEW

**Comment by zwangsheng** [comment]
friendly ping @pan3793 @turboFei

File: kyuubi-server/src/test/scala/org/apache/kyuubi/server/api/v1/BatchesResourceSuite.scala [hunk 1]
```diff
@@ -576,23 +576,40 @@ abstract class BatchesResourceSuiteBase extends KyuubiFunSuite
...
+        eventually(timeout(10.seconds)) {
+          assert(session1.batchJobSubmissionOp.getStatus.state === OperationState.RUNNING ||
+            session1.batchJobSubmissionOp.getStatus.state === OperationState.FINISHED)
+          assert(session1.batchJobSubmissionOp.builder.processLaunched)
+
+          assert(session2.batchJobSubmissionOp.getStatus.state === OperationState.RUNNING ||
+            session2.batchJobSubmissionOp.getStatus.state === OperationState.FINISHED)
+          assert(!session2.batchJobSubmissionOp.builder.processLaunched)
+        }
+      case "2" =>
```
Comment (pan3793) [inline kyuubi-server/src/test/scala/org/apache/kyuubi/server/api/v1/BatchesResourceSuite.scala:599]: the check for 2 seems much lenient than 1, can we make it strict?

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala [hunk 2]
```diff
@@ -140,13 +144,37 @@ class KyuubiRestFrontendService(override val serverable: Serverable)
...
+  }
+
   @VisibleForTesting
   private[kyuubi] def recoverBatchSessions(): Unit = {
     val recoveryNumThreads = conf.get(METADATA_RECOVERY_THREADS)
     val batchRecoveryExecutor =
       ThreadUtils.newDaemonFixedThreadPool(recoveryNumThreads, "batch-recovery-executor")
     try {
-      val batchSessionsToRecover = sessionManager.getBatchSessionsToRecover(connectionUrl)
+      // split metadata wait to recover as v1 and v2
```
Comment (pan3793) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala:158]: the code is clear enough to show **what we are doing**, the comment should explain "why should we do it"

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala [hunk 3]
```diff
@@ -140,13 +144,37 @@ class KyuubiRestFrontendService(override val serverable: Serverable)
     batchChecker.scheduleWithFixedDelay(task, interval, interval, TimeUnit.MILLISECONDS)
   }
 
+  private def batchV2Enabled(reqConf: Map[String, String]): Boolean = {
+    KyuubiServer.kyuubiServer.getConf.get(BATCH_SUBMITTER_ENABLED) &&
+    reqConf.getOrElse(BATCH_IMPL_VERSION.key, getConf.get(BATCH_IMPL_VERSION)) == "2"
+  }
```
Comment (pan3793) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala:150]: copied from another place? can we make the method sharable?

**Pull request closed without merging**